drivers/at86rf2xx/at86rf2xx_rf_ops.c

```diff
@@ -0,0 +1,971 @@
+    return 0;
+}
+static int _request_set_rx(at86rf2xx_t *dev)
+{
+    DEBUG("at86r2xx_rf_ops: request_set_rx\n");
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_PHY_STATE_RX);
+    return 0;
+}
+
+static int _request_set_tx(at86rf2xx_t *dev, bool force)
+{
+    DEBUG("at86r2xx_rf_ops: request_set_tx\n");
+    /* Prevent issue described in https://github.com/RIOT-OS/RIOT/pull/11256 */
+    if (at86rf2xx_get_status(dev) == AT86RF2XX_PHY_STATE_RX_BUSY && !force) {
+        return -EBUSY;
+    }
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_TRX_STATE__FORCE_PLL_ON);
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_PHY_STATE_TX);
+    return 0;
+}
+
+static int _request_cca(at86rf2xx_t *dev)
+{
+    DEBUG("at86rf2xx_rf_ops: request_cca\n");
+    uint8_t reg;
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE, AT86RF2XX_TRX_STATE__FORCE_PLL_ON);
+    at86rf2xx_set_state(dev, AT86RF2XX_STATE_RX_ON);
+
+    reg = at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_MASK);
+    reg |= AT86RF2XX_IRQ_STATUS_MASK__CCA_ED_DONE;
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__IRQ_MASK, reg);
+    /* Perform CCA */
+    reg = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_CC_CCA);
+    reg |= AT86RF2XX_PHY_CC_CCA_MASK__CCA_REQUEST;
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__PHY_CC_CCA, reg);
+    return 0;
+}
+
+static int _request_op(ieee802154_dev_t *hal, ieee802154_hal_op_t op, void *ctx)
+{
+    at86rf2xx_t *dev = hal->priv;
+
+    int res = -ENOTSUP;
+    mutex_lock(&dev->lock);
+    switch (op) {
+    case IEEE802154_HAL_OP_TRANSMIT:
+#if AT86RF2XX_HAVE_RETRIES && AT86RF2XX_IS_PERIPH
+        dev->tx_retries = -1;
+#endif
+        /* trigger sending of pre-loaded frame */
+        at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_STATE,
+                             AT86RF2XX_TRX_STATE__TX_START);
+        res = 0;
+        break;
+    case IEEE802154_HAL_OP_SET_RX:
+        res = _request_set_rx(dev);
+        break;
+    case IEEE802154_HAL_OP_SET_IDLE:
+        res = _request_set_tx(dev, *((bool*) ctx));
+        break;
+    case IEEE802154_HAL_OP_CCA:
+        res = _request_cca(dev);
+        break;
+    default:
+        assert(false);
+        break;
+    }
+    mutex_unlock(&dev->lock);
+    return res;
+}
+
+static int _confirm_transmit(at86rf2xx_t *dev, ieee802154_tx_info_t *info)
+{
+DEBUG("at86rf2xx_rf_ops: confirm_transmit\n");
+    uint8_t status = at86rf2xx_get_status(dev);
+    if (status == AT86RF2XX_STATE_BUSY_TX_ARET || status == AT86RF2XX_STATE_BUSY_TX) {
+        return -EAGAIN;
+    }
+
+    if (info) {
+        uint8_t trac_status = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_STATE) & AT86RF2XX_TRX_STATE_MASK__TRAC;
+#if AT86RF2XX_HAVE_RETRIES && AT86RF2XX_HAVE_RETRIES_REG
+        info->retrans = (at86rf2xx_reg_read(dev, AT86RF2XX_REG__XAH_CTRL_2)
+                       & AT86RF2XX_XAH_CTRL_2__ARET_FRAME_RETRIES_MASK) >>
+                      AT86RF2XX_XAH_CTRL_2__ARET_FRAME_RETRIES_OFFSET;
```

Comment:
```suggestion
        info->retrans = (at86rf2xx_reg_read(dev, AT86RF2XX_REG__XAH_CTRL_2)
                         & AT86RF2XX_XAH_CTRL_2__ARET_FRAME_RETRIES_MASK)
                        >> AT86RF2XX_XAH_CTRL_2__ARET_FRAME_RETRIES_OFFSET;
```

hmm this still doesn't look nice, but maybe a little less confusing?